docs(memfault): suggest Memfault in NCS docs (#23)
memfault/sdk-nrf: doc/nrf/test_and_optimize/debugging.rst

For more information about monitor-mode debugging, see Zephyr's :ref:`zephyr:debugmon` documentation and SEGGER's `Monitor-mode Debugging <Monitor-mode Debugging_>`_ documentation.

> Remote Debugging
> ******************************

Reviewer: ```suggestion
Remote debugging using Memfault
*******************************
```

> The |NCS| provides remote observability support to maintain optimal device performance
> and reliablity in the midst of ever-changing device ecosystems. As you start to rollout your fleet,
> the nRF Connect SDK enables developers to debug and monitor devices remotely.

Reviewer: ```suggestion
  The |NCS| provides remote observability support to maintain optimal device performance and reliability during the ever-changing device ecosystems. 
  As you start to rollout the fleet, you can debug and monitor devices remotely.
```

Author: I updated this to match what we decided on for the [remote observability description](https://github.com/memfault/sdk-nrf/pull/23#discussion_r1598433927)

> This functionality allows you to efficiently collect crash information and get the same information you would see at
> your desk with a debugger from devices half way around the world!

Reviewer: ```suggestion
This functionality allows you to efficiently collect crash information and get the same information as the other debugger.
```

Author: I think referring to `the other debugger` might be ambiguous. What do you think of: 

```suggestion
This functionality allows you to efficiently collect crash information and get the same data from remote devices as you would when using a debugger.
``````

Reviewer: sounds good.
